alexandria/core/models.py

```diff
@@ -144,17 +151,34 @@ class Document(UUIDModel):
     marks = models.ManyToManyField(Mark, blank=True, related_name="documents")
     date = models.DateField(blank=True, null=True)
 
+    @transaction.atomic()
     def clone(self):
         latest_original = (
             self.files.filter(variant="original").order_by("-created_at").first()
         )
         self.pk = None
         self.save()
 
-        # TODO: decide on how to deal with thumbnail creation
-        latest_original.pk = None
-        latest_original.document = self
-        latest_original.save()
+        with NamedTemporaryFile() as tmp:
+            temp_file = Path(tmp.name)
+            with temp_file.open("w+b") as file:
+                file.write(latest_original.content.file.file.read())
+
+                latest_original.content = DjangoFile(file)
+                latest_original.pk = None
+                latest_original.document = self
+                latest_original.save()
+
+        from alexandria.core.utils import create_thumbnail
+
+        try:
+            create_thumbnail(latest_original)
+        except DjangoCoreValidationError as e:
+            log.error(
+                "Object {obj} created successfully. Thumbnail creation failed. Error: {error}".format(
+                    obj=latest_original, error=e.messages
+                )
+            )
```

Comment:
I think this should respect the `ALEXANDRIA_ENABLE_THUMBNAIL_GENERATION` setting.

Reply:
Nice catch! I put the condition directly into the `create_thumbnail` function so we don't have to think about it on every callsite.